### PR TITLE
Staking migration updates

### PR DESCRIPTION
**Pull request by mikestarrdev**

### CODE REVIEW

File: src/features/staking/utils/convertBigIntToBigNumber.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import BigNumber from "bignumber.js";
+
+export const convertBigIntToBigNumber = (balance: bigint | number) => {
```
Comment (gpxl-dev) [inline src/features/staking/utils/convertBigIntToBigNumber.ts:3]: This function is badly named - it's returning a `number` and not `bigint` or `BigNumber`.
I think I would call it something like `bigIntToUnits`, and not accept a `number` as a param.

File: src/features/staking/utils/convertUnixToDays.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+export const convertUnixToDays = (maturity: bigint | undefined) => {
```
Comment (gpxl-dev) [inline src/features/staking/utils/convertUnixToDays.ts:1]: I tend to use `date-fns` for this, but DIY is also fine :)

File: src/features/staking/StakingModal.tsx [hunk 3]
```diff
@@ -22,7 +23,7 @@ export const StakingModal = () => {
 
   const formReturn = useForm();
   const { getValues } = formReturn;
-  const stakingAmountFormatted = getValues().stakingAmount;
+  const stakingAmount = getValues().stakingAmount * 10 ** 4 || 0;
```
Comment (gpxl-dev) [inline src/features/staking/StakingModal.tsx:26]: Make this a bigint.

`BigInt(getValues().stakingAmount || 0n) ** (10n ** 4n)`

**Review by gpxl-dev — CHANGES_REQUESTED** [review]
I haven't had time to actually test this out yet, but the logic looks great.

Just one "Error" really (which will likely never cause any issues due to AST's decimals, but a big whale could break things) - atomic amounts should always be bigints - you are safe to use number after you've divided by decimals but not after you've multiplied it back up again.

File: src/features/staking/StakingModal.tsx (hunk 3 above)
Comment (mikestarrdev) [reply src/features/staking/StakingModal.tsx:26]: The issue I'm facing with this is that when a floating number is passed in, trying to convert it to a `BigInt` causes the app to crash

File: src/features/staking/utils/convertUnixToDays.ts (hunk 2 above)
Comment (mikestarrdev) [reply src/features/staking/utils/convertUnixToDays.ts:1]: At this point I'll keep the DIY solution since it's already there

**Review by gpxl-dev — APPROVED** [review]

**Review by gpxl-dev — APPROVED** [review]

**Pull request merged**